### DISCUSSION TITLE
Video/help

### DISCUSSION THREAD

**Discussion by TheHillsa** [subdavis/kobo-book-downloader#72] [Q&A] [answered]
Hi everyone. Can somebody create a video on how to use this? it would help noobs like me.

I downloaded a pre-built bundle for windows as it does not require python. Ran it, and nothing happened. No directories were created, I cannot find any kobo files related to the install.

None of the commands work in cmd either.

I am not sure how to continue. Thanks for any help.

**Comment by subdavis** [comment, marked as answer]
Can you copy and paste everything in your CMD or Powershell when you run the application?

1. Open CMD
2. Run `./kobodl.exe`
3. See some output.

Copy all the text, including the command you ran and what was printed back.

**Reply by TheHillsa** [reply]
Executing it through CMD worked. It now gives me:

Options:
  --fmt TEXT     python-tabulate table format string
  --config FILE  path to kobodl.json config file
  --debug        enable the debug log
  --version      Show the version and exit.
  --help         Show this message and exit.

Commands:
  book   list and download books
  serve  start an http server
  user   show and create users

But the instructions on what to do next are still pretty complex for me. For example if I type "user", it gives me this: "'user' is not recognized as an internal or external command, operable program or batch file."

**Reply by subdavis** [reply]
So it sounds like you're new to the command line.   That's OK!  However, the user guide in the documentation does tell you what to do next.

https://github.com/subdavis/kobo-book-downloader#usage

These docs are written for all platforms and install methods, but for you, replace `kobodl` with `./kobodl.exe`

For example:

```bash
# Add a new user
./kobodl.exe user add

# List books
./kobodl.exe book list

# Download ALL books with default options when only 1 user exists
./kobodl.exe book get --get-all
```

**Reply by subdavis** [reply]
And because I like to improve the documentation every time someone has a question like this, Ive added a note to https://github.com/subdavis/kobo-book-downloader/blob/main/README.md#usage -- I hope this helps.

**Reply by TheHillsa** [reply]
Yes, that is great help. The instructions are excellent and easy to follow. While attempting to set it up, I ran into a problem. When I copy/pasted password into cmd, nothing entered. However, if I type something else upon confirmation, I got the "Error: the two entered values do not match", so it was safe to assume that the password was being typed in, but it was hidden. However upon copy/pasting the accurate password both times and following the next steps to getting the recaptcha, I got the login failure error. Before asking for help though, I entered the password in manually (which was a pain as I use generated passwords which tend to be complex), and it worked. So in case this happens to anyone else, entering the password manually is the solution.

But there is another issue that I do not know how to solve. When I use the command: 

`kobodl-windows.exe book get \ --user test123123312@google.com \ --output-dir E:/Kobo \ --format-str '{Title}' \ --get-all`

I get:

```
error: cannot pass product IDs when --get-all is used. Use one or the other.
Traceback (most recent call last):
  File "kobodl\__main__.py", line 5, in <module>
  File "click\core.py", line 829, in __call__
  File "click\core.py", line 782, in main
  File "click\core.py", line 1259, in invoke
  File "click\core.py", line 1259, in invoke
  File "click\core.py", line 1066, in invoke
  File "click\core.py", line 610, in invoke
  File "click\decorators.py", line 33, in new_func
  File "kobodl\commands\book.py", line 73, in get
NameError: name 'exit' is not defined
[16960] Failed to execute script '__main__' due to unhandled exception!
```

Using:

`kobodl-windows.exe book list --read`

Gives me:

```
Title                            Author                         RevisionId                            Owner
-------------------------------  -----------------------------  ------------------------------------  ------------------------
Collateral Damage (🎧 Audiobook)  Laila Al-Arian & Chris Hedges  ba09e0a9-8cd7-42c7-8c91-04d584765a79  test123123312@google.com
```

**Reply by subdavis** [reply]
The `\` character is there because of the line wraps.  If you aren't entering multi-line text, remove all the `\` characters. The tool is interpreting that as input.

**Reply by TheHillsa** [reply]
> The `\` character is there because of the line wraps. If you aren't entering multi-line text, remove all the `\` characters. The tool is interpreting that as input.

Thank you very much. It worked perfectly. It definitely isn't obvious to noobs like me, I think it could be worthwhile to add to the documentation. The error "cannot pass product IDs when --get-all is used. **Use one or the other.**" is also not obvious what it means by one or the other.

This audiobook is about the collateral damage caused by war. It interviews veterans (those who had tours in Iraq), who share the practices of the army (e.g. "Military convoys traveling at tremendous speeds through towns have become trains of death. Civilians are routinely run over or shot to death. Soldiers fire upon Iraqi vehicles with impunity at checkpoints. Late-night detentions based on shoddy intelligence terrify women, traumatize children, and radicalize the young men caught in their dragnet"), and their experiences. War isn't an abstract thing, It causes a ton of death and suffering. So it is important to understand the extent of such decisions. If you are interested, let me know and I'll upload it!
